coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/CoordinatorRuntime.java

```diff
@@ -1378,7 +1383,7 @@ public void complete(Throwable exception) {
 
             if (deferredEventQueuedTimestamp != NOT_QUEUED) {
                 // Only record the purgatory time if the event was deferred.
-                runtimeMetrics.recordEventPurgatoryTime(purgatoryTimeMs);
+                runtimeMetrics.recordEventPurgatoryTime(Math.min(MAX_LATENCY_MS, purgatoryTimeMs));
```

Comment:
In my opinion, this gating should not be in the runtime because it depends on the underlying implementation of the metric. Could we move it into the runtime metrics implementation?